x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/Layout.java

```diff
@@ -108,16 +108,27 @@ public Layout build() {
             int numberOfChannels = 0;
             for (ChannelSet set : channels) {
                 int channel = numberOfChannels++;
-                for (NameId id : set.nameIds) {
-                    ChannelAndType next = new ChannelAndType(channel, set.type);
-                    ChannelAndType prev = layout.put(id, next);
-                    // Do allow multiple name to point to the same channel - see https://github.com/elastic/elasticsearch/pull/100238
-                    // if (prev != null) {
-                    // throw new IllegalArgumentException("Name [" + id + "] is on two channels [" + prev + "] and [" + next + "]");
-                    // }
+                if (set != null) {
+                    for (NameId id : set.nameIds) {
+                        ChannelAndType next = new ChannelAndType(channel, set.type);
+                        ChannelAndType prev = layout.put(id, next);
+                        // Do allow multiple name to point to the same channel - see https://github.com/elastic/elasticsearch/pull/100238
+                        // if (prev != null) {
+                        // throw new IllegalArgumentException("Name [" + id + "] is on two channels [" + prev + "] and [" + next + "]");
+                        // }
+                    }
                 }
             }
             return new DefaultLayout(Collections.unmodifiableMap(layout), numberOfChannels);
         }
+
+        public void replace(NameId id, NameId id1) {
```

Comment:
Why is this needed by mv_expand and not other commands?

Reply:
> Maybe mv_expand needs to preserve the original attributes instead?

It was like this at the beginning, but that's not correct (see [wrong folding](https://github.com/elastic/elasticsearch/issues/101118)).

Before this PR, the layout was rebuilt from scratch replacing the expanded field, but it didn't work properly in cases like this, where the layout.inverse() contains null entries (because multiple projections point to the same layout position, or at least this is my understanding).

This change just makes the layout planing a bit more straight-forward and less error prone, but it doesn't change the previous logic.

> Why is this needed by mv_expand and not other commands?

mv_expand implementation is pretty peculiar since the beginning: it assumed that the input and output layouts were the same (that is not completely true, se above)